using_restricted_array.rb

```diff
@@ -96,11 +96,12 @@ def binary_search(array, length, value_to_find)
     if value_to_find == array[mid]
       return true
     elsif value_to_find > array[mid]
-      i = mid
+      i = mid + 1
     else #value_to_find < array[mid]
-      j = mid
+      j = mid - 1
     end
   end
+  return false 
```

Comment:
Thank you for updating the code based on the updated algorithm. There is one more change needed. Please take a look at the updated binary_search slide. 

Here's the considerations I was missing in the algorithm on the slide before the update: (1) what if there is only one element in the array and it is the value_to_find? In this case, it used to return false. (To fix this, add an additional if statement after the loop ends, to check if array[low] is the value_to_find.) (2) Consider the scenario where there are two elements in the array and the value_to_find is the second element in the array. In the first iteration, min = 0, max=1, mid=0. Value is not at index 0. So, min = mid (still 0). On re-entering the loop, mid would be still 0. So we had an infinite loop. (To fix this, update max to be mid -1, if value_to_find is less than mid. And update min = mid+1, if value_to_find is greater than mid.)